publication-commons/src/main/java/no/unit/nva/publication/service/impl/UpdateResourceService.java

```diff
@@ -164,9 +170,69 @@ private ArrayList<TransactWriteItem> createTransactions(Resource resource, UserI
         transactionItems.addAll(updateFilesTransactions(resource, userInstance, persistedResource, importSource));
         transactionItems.addAll(ticketsTransactions);
         transactionItems.addAll(updatePublicationChannelsForPublisherWhenDegree(resource, persistedResource));
+        transactionItems.addAll(createResourceRelationTransactions(persistedResource, resource));
+        transactionItems.addAll(createRefreshRelatedResourcesTransactions(persistedResource));
         return transactionItems;
     }
 
+    // TODO: After PK migration to be based on resource identifier only: update version using partial update without fetching resources
+    private List<TransactWriteItem> createRefreshRelatedResourcesTransactions(Resource persistedResource) {
+        var relatedResources = persistedResource.getRelatedResources();
+        if (!relatedResources.isEmpty()) {
+            return relatedResources.stream()
+                       .map(readResourceService::getResourceByIdentifier)
+                       .filter(Optional::isPresent)
+                       .map(Optional::get)
+                       .map(this::createPutTransaction)
+                       .toList();
+        }
+        return Collections.emptyList();
+    }
```

Comment:
When PK0 and SK0 will be based on resource identifier only, there will be no need to fetch resource before doing "update version" operation.